src/modules/pmc/src/lib/PmcBase.h

```diff
@@ -36,14 +42,45 @@ class PmcLog
 class PmcBase
 {
 public:
+    struct DesiredState
+    {
+        std::vector<std::string> packages;
+    };
+
+    struct State
+    {
+        ExecutionState executionState;
+        std::string packagesFingerprint;
+        std::vector<std::string> packages;
+    };
+
     PmcBase(unsigned int maxPayloadSizeBytes);
     virtual ~PmcBase() = default;
 
     static int GetInfo(const char* clientName, MMI_JSON_STRING* payload, int* payloadSizeBytes);
     virtual int Set(const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes);
     virtual int Get(const char* componentName, const char* objectName, MMI_JSON_STRING* payload, int* payloadSizeBytes);
     virtual unsigned int GetMaxPayloadSizeBytes();
+    virtual int RunCommand(const char* command, bool replaceEol, std::string* textResult, unsigned int timeoutSeconds) = 0;
 
 private:
+    bool CanRunOnThisPlatform();
+    int ExecuteUpdate(const std::string& value);
+    int ExecuteUpdates(const std::vector<std::string> packages, bool updatePackageLists);
+    std::string GetFingerprint();
+    std::vector<std::string> GetReportedPackages(std::vector<std::string> packages);
+    int DeserializeDesiredState(rapidjson::Document& document, DesiredState& object);
+    int ValidateAndGetPackagesNames(std::vector<std::string> packagesLines);
+    static std::vector<std::string> ListFiles(const char* dir, const char* fileNameExtension);
+    static int SerializeState(State reportedState, MMI_JSON_STRING* payload, int* payloadSizeBytes, unsigned int maxPayloadSizeBytes);
+    static int CopyJsonPayload(rapidjson::StringBuffer& buffer, MMI_JSON_STRING* payload, int* payloadSizeBytes);
+    static std::vector<std::string> Split(const std::string& str, const std::string& delimiter);
+    static std::string TrimStart(const std::string& str, const std::string& trim);
+    static std::string TrimEnd(const std::string& str, const std::string& trim);
+    static std::string Trim(const std::string& str, const std::string& trim);
```

Comment:
>str [](http://example.com/codeflow?start=47&length=3)

Use full words. No "str" please. Use "value" or something else.

Reply:
Applied the change, thank you! Could you consider, in the future, to move the helper methods for string handling in a common class? We have duplicated code with the HostName module.